### ISSUE TITLE
Question: Why does componentDidUpdate use prevProps when checking this.isLazy?

### ISSUE THREAD

**Issue by shanecav** [jamesplease/react-request#171]
(related to #59)

https://github.com/jamesplease/react-request/blob/5eddd22c7e0c2bbd3bfde49effad33eb6b512c08/src/fetch.js#L124

Is there a reason for calling `this.isLazy(prevProps)` as opposed to `this.isLazy()` (i.e. letting it default to `this.props`)?

I ask because if it checked current `this.props` for "laziness", I think you'd be able to do basic serial requests with something like this:

```js
import Composer from "react-composer"

<Composer components={[
  <Fetch url="https://someapi.com/blah" />,
  ({ results: [something], render }) => (
    <Fetch
      lazy={!something.data}
      url={`https://someapi.com/${something.data}`}
      children={render}
    />
  )
}] />
```

i.e. the 2nd request, which depends on `something.data`, wouldn't go out until `something.data` existed.

As it is now, the `this.isLazy` check in `componentDidUpdate` is one render behind, so the above isn't possible. Just curious if there's a reason for that I'm not seeing.

P.S. If there's a simpler/better way of doing serial requests, I'd love to hear it! I know that's something you're working on hashing out right now - I'm definitely curious what you're thinking.

Thanks!

**Comment by jamesplease** [comment]
:wave: hi again @shanecav ! Thanks again for opening another great issue!

> Is there a reason for calling this.isLazy(prevProps) as opposed to this.isLazy()

This is actually a bug that was introduced in 2.0.4. When I switched from `componentWillReceiveProps` (in which case I was passing `nextProps`) to `componentDidUpdate`, I should have gotten rid of the usage of `nextProps` altogether rather than switching it to `prevProps`.

This will be fixed in 3.0.1 in a few minutes. I opened a PR over at #172 .

I really appreciate for your thoroughness!
